### ISSUE TITLE
Random Freezes

### ISSUE THREAD

**Issue by godleydemon** [psi-plus/psi-plus-snapshots#4]
Having a weird problem with version 0.16.361, client freezes up every 5 minutes or so and stays frozen for about a minute or so then comes back just fine. Could be the amount of users I have on my contacts list; currently 924 contacts.

Let me know if there is any debug information I can give you, because currently I don't know where any debug stuff is for Psi+

**Comment by rapgro** [comment]
Revision 361 seems to be very old. Maybe retry with a current snapshot.

**Comment by godleydemon** [comment]
Well it's actually the newest precompiled version I could find, I actually couldn't quite figure out how to compile the windows version of this and actually tried for a little while to do so.

**Comment by rapgro** [comment]
Sorry, I can not help with Windows.

**Comment by godleydemon** [comment]
hmm.. where can I go to get some help with windows then? Because the freezing is awfully annoying, other then that, I love psi+ XD lol

**Comment by Vitozz** [comment]
Freezes occur because of the protocol ipv6 (on Windows). Psi+ first tries to connect to the server by ipv6 protocol, and can not do this (usually) and waits for about a minute until the attempt fails, then connected on ipv4. To solve this problem try to manually set IP-address of the server in the account settings or wait a little (very soon this problem will be fixed, maybe it's already fixed).
And also this repo is only for snapshots for package-makers for linux systems. All issues must be placed in main repo "psi-plus/main"
Sorry for my english (google translate).

**Comment by Ri0n** [comment]
It is another problem and it's not yet fixed in master branch.

I can't say anything about periodic freezes but xml logs could be useful.

10 мая 2015 г. 17:20 пользователь "Vitaly Tonkacheyev" <
notifications@github.com> написал:
>
> Freezes occur because of the protocol ipv6 (on Windows). Psi+ first tries
to connect to the server by ipv6 protocol, and can not do this (usually)
and waits for about a minute until the attempt fails, then connected on
ipv4. To solve this problem try to manually set IP-address of the server in
the account settings or wait a little (very soon this problem will be
fixed, maybe it's already fixed).
> And also this repo is only for snapshots for package-makers for linux
systems. All issues must be placed in main repo "psi-plus/main"

**Comment by godleydemon** [comment]
If you tell me exactly where to find the xml logs, I'll throw the logs up on here once it freezes again today. But the problem with IPv6 on this PC is that it's disabled http://puu.sh/hIFIS/73ce4b4269.png . It had to be disabled since it acts funky with some proprietary software from work. So I'm not quite sure if it's actually IPv6 causing the problem.

**Comment by Ri0n** [comment]
right click on account in your contact list window.

you have to remember stanzas before and after freeze. so keep xml console
open and activated.

2015-05-11 2:27 GMT+05:00 godleydemon <notifications@github.com>:

**Comment by godleydemon** [comment]
I had to clip it a little at the top, so I tried to get it around abouts when it actually froze up. But what ended up happening was, I looked at the service discovery to get to a specific room I don't have bookmarked and it froze up. After a minute or so it unfroze and spit out nearly every contact into the xml console logging there status changes. 

https://gist.github.com/godleydemon/f6070373736874cc5e53

**Comment by Ri0n** [comment]
service discovery  could be quite long on big lists.
but I'm agreed GUI should not be blocked while receiving/parsing data. One
day we will fix it.

2015-05-11 3:02 GMT+05:00 godleydemon <notifications@github.com>:

**Comment by godleydemon** [comment]
The other freezes also happen during regular day to day things. I'm thinking it's just the massive amounts of contacts I have, with the psi+ logging all of those changes it's locking up and trying to catch up.

**Comment by godleydemon** [comment]
<iq type="result" id="249--1061944455"/>
Just occurred in my xml console as it froze up again.

**Comment by wadealer** [comment]
godleydemon, you can find latest windows bulds here: https://sourceforge.net/projects/psiplus/files/MS-Windows  or here (recomended): https://yadi.sk/d/FD8HHORTQ5HYc

**Comment by godleydemon** [comment]
Looking at the nightly builds, when I try to run this or follow the directions that is included, it never works and always throws errors at me. I'm trying the build on the yadi site that was release 01-21-2015 to see if that works or not. I'll update to let you know what happens.

**Comment by godleydemon** [comment]
ok looks like the build is working, but I won't know for sure if it's fixed the problem until tomorrow when I go back to work and start using it heavily again. Though this usually happens without being used heavily as well, I don't usually notice it.

**Comment by godleydemon** [comment]
Sorry to leave a message so quickly again here, but it just happened again while I was in a chat room. I didn't have the xml logger open at the time so I don't have a log, though something curious did happen with the luna theme with this version of psi+ http://puu.sh/hJ9sz/7047923123.jpg

**Comment by wadealer** [comment]
Non-webkit to webkit migration probably. 
Check if your issue will reproduced on new profile.

**Comment by godleydemon** [comment]
yeah I have a separate profile that this happens on, I've also went through and wiped out psi+ and used a cleanup utility to try and wipe all traces of it from the system and reinstall it utilizing the recommended build from the psi+ website. I just don't know why it's happening lol.
